tests/SchemaTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace tests\Libero\Schemas;
+
+use DOMDocument;
+use FluentDOM;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Finder\Finder;
+
+final class SchemaTest extends TestCase
```

Comment:
Doesn't matter what language this is done in. Both PHP and Python would use libxml2, where the error messages aren't great. Using [Jing](https://github.com/relaxng/jing-trang) would be better (Oxygen uses it), but means introducing Java. (Not too adverse to it though, since it's purely for the test suite.)

Reply:
If PHP/Python were unable to validate the schema effectively (or produce huge error messages), it would be a problem for implementing services; hence dogfooding the choice here makes sense.